### ISSUE TITLE
CMake caching of OpenMP flags is broken

### ISSUE THREAD

**Issue by esiragusa** [seqan/seqan#1481]
CMake now tries to detect OpenMP flags for each app, test folder and demo that needs them. 

```
-- Try OpenMP CXX flag = [ ]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [/openmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [-Qopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [-openmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [-xopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [+Oopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [-qsmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Try OpenMP CXX flag = [-mp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Failed
-- Could NOT find OpenMP (missing:  OpenMP_CXX_FLAGS) 
```

Note that this thing is particularly slow for Xcode builds. Before this lookup was done only once and cached.

**Comment by h-2** [comment]
We have removed a bunch of cmake code before the release, because it was duplicated system code that caused problems in different places.
Do you have an idea for caching that does not involve maintain modified copies of system code?

**Comment by esiragusa** [comment]
Can't you have a look at the code that you removed?

**Comment by h-2** [comment]
I know how it worked, but we completely removed the obscure seqan-invented dependency tracking, so it can't be replaced by the old code. The only thing I can think of right now is moving the dependency check out of the apps entirely for regular builds, but it's not something I am keen on doing. IMHO we need to remove seqan build system cruft and make the apps self-contained for later removal from the tree.

**Comment by h-2** [comment]
I guess I can work something out, let me have a look...

**Comment by esiragusa** [comment]
Ah if you plan on moving all apps out of seqan in the very short term, then spending a lot of time to fix this issue is not worth. If it's hard to fix, never mind. That's not crictical... Thanks.

**Comment by h-2** [comment]
@xenigmax  I will try to do this in the next, but if it doesn't get done for the release it's not a problem.

**Comment by h-2** [comment]
will be fixed by #1560

**Comment by marehr** [comment]
We may have to look at it again, because we had some performance degrade on windows.

see #1586

**Comment by h-2** [comment]
solved by #1184
